src/cron/generateInvites.ts

```diff
@@ -0,0 +1,55 @@
+import fastq from 'fastq';
+import { Cron } from './cron';
+import { Feature, FeatureValue, Invite } from '../entity';
+import { generateUUID } from '../ids';
+
+const cron: Cron = {
+  name: 'generate-invites',
+  handler: async (con, logger) => {
+    const featuresQuery = con
+      .createQueryBuilder()
+      .from(Feature, 'f')
+      .where(
+        'f."createdAt" < now() - interval \'7 days\' AND f."value" = :value',
+        { value: FeatureValue.Allow },
+      ).andWhere(`
```

Comment:
I'm not sure if there is an alternative, but think this was pretty costs to run as a query.
Will need to dive into other options for you.

Reply:
an alternative would be to use a `left join` here, but given that we are indexing both `userId` and `campaign`, using a select might actually be more performant. I'm open to other suggestions though

Reply:
If it's indexed properly I don't think it should be costly. Another option is not to query for existing invites and on insert just ignore conflicts. I don't mind either, we just need to make sure we have proper indexes

Reply:
The relevant columns are indexed: https://github.com/dailydotdev/daily-api/blob/e1672479f90a7d5978610795fb05e31447c43921/src/migration/1697708253831-Invites.ts#L10-L12 so we should be good here 👍